rfcs/text/remove-support-for-uncontrolled-inputs.md

```diff
@@ -0,0 +1,224 @@
+1. Decimal
+1. Number Input
+1. Simple Select
+1. Filterable Select
+1. Multi Select
+1. Grouped Character
+1. Numeral Date
+1. Search
+
+# Unit Testing
+
+Currently uncontrolled and controlled behaviours are unit tested. As part of this work we could reduce the amount of code in the spec files quite considerably. If we take `Decimal` as an example of this, `Decimal` currently has 1430 lines of code in the spec file. This could be reduced to down to around 480 lines of code if we were to remove the uncontrolled tests. This would make the tests easier to read and maintain.
+
+# Drawbacks
+
+This work will be a `BREAKING CHANGE` however we will communicate this in the appropriate Carbon channels. Consumers of Carbon will also have the opportunity to comment on this RFC.
+
+# Alternatives
+
+Do nothing; we leave the input components as they are and continue to support controlled and uncontrolled behaviours. This course of action would mean no impact to our consumers. Uncontrolled input components can be used with Formik, even though this is not encouraged, through implementing a store type (React's [useReducer hook](https://reactjs.org/docs/hooks-reference.html#usereducer) or [Redux](https://redux.js.org/) for example) mechanism in order to retrieve a value. However, we would always still recommend using controlled input components with Formik.
+
+# Adoption strategy
+
+As mentioned, this proposed work will be implemented as a `BREAKING CHANGE`. In terms of how consuming projects will upgrade Carbon, we will need to document the upgrade process and link to it in the release notes. If a project has used an input as uncontrolled, they will need to update their codebase to make this controlled instead. This will involve making sure that the value is stored using React's [useState](https://reactjs.org/docs/hooks-state.html) hook to retain the value and making sure this is managed using an `onChange` function. We may have to provide additional support in this scenario to provide examples of how they can achieve the same functionality using this approach.
+
+We also already provide various codemods in our [Carbon Codemod](https://github.com/Sage/carbon-codemod) repo that could be used by our consumers to assist with the move to a controlled component. Examples of this are:
+- A [rename-prop](https://github.com/Sage/carbon-codemod/tree/master/transforms/rename-prop) codemod that consumers could use to change a `defaultValue` value prop to `value`.
+- A [add-prop](https://github.com/Sage/carbon-codemod/tree/master/transforms/add-prop) codemod that could be used to add to a `value` prop.
+- A [remove-prop](https://github.com/Sage/carbon-codemod/tree/master/transforms/remove-prop) codemod that could be used to remove a `defaultValue` prop.
+- A [replace-prop-value](https://github.com/Sage/carbon-codemod/tree/master/transforms/replace-prop-value) codemod that changes the value of a prop.
+
+There is also a lot of support for writing controlled components.
```

Comment:
Where?

Reply:
That is a valid question. I shall make this more specific 👍🏻